### ISSUE TITLE
Process expansion stopped working on OS X

### ISSUE THREAD

**Issue by waterhouse** [fish-shell/fish-shell#2386]
On current master, when I try to use process expansion by name, e.g. "%firefox", it says:
```
Unable to find a process 'firefox'
fish: echo %firefox
            ^
```
In fact, the only process that it can find is %self.

I did a git-bisect to find where the problem was introduced.  There were three commits on which the build failed with a "features.h: file not found" error.  But it did knock it down to a set of four commits.
```
~/fish-shell> git bisect skip
There are only 'skip'ped commits left to test.
The first bad commit could be any of:
793e1afa084982dac92c4fe19e50c25e326a79c2
3929e9de0e69666b37df87347d5ce15663e81347
78162a2a1343a7caff3d43f457daa87475e6b926
958880d97251b09af5c639a11d914067929e91af
We cannot bisect more!
```
One of these, 3929e9de0e69666b37df87347d5ce15663e81347, did make a bunch of changes to process expansion, so I suspect the culprit is in there.

**Comment by faho** [comment]
`fish --version`? Your operating system?

For the record, this works for me with "fish, version 2.2.0-257-g4bc8cf1", i.e. current master, on archlinux.

**Comment by waterhouse** [comment]
Right, yes, forgot to mention that.  Operating system is OS X Yosemite, 10.10.5.  (The same problem occurs on OS X 10.6.8.)
```
~/fish-shell> ./fish --version
fish, version 2.2.0-257-g4bc8cf1
~/fish-shell> ./fish -c 'echo %firefox'
Unable to find a process 'firefox'
fish: echo %firefox
            ^
~/fish-shell> ./fish -c 'echo %fish'
Unable to find a process 'fish'
fish: echo %fish
            ^
```

**Comment by faho** [comment]
How did you build fish? Maybe there's something wrong there.

I don't have an OSX machine so if it's truly specific to that I can't test it, but @ridiculousfish and @zanchey do.

**Comment by waterhouse** [comment]
```
~/fish-shell> git reset --hard origin/master
HEAD is now at a02d4dc Document cross-product
 ~/fish-shell> ls
CONTRIBUTING.md             README.md                   doc_src                     osx
COPYING                     build_tools                 etc                         po
Dockerfile                  config.guess                fish.pc.in                  release_notes.html
Doxyfile                    config.sub                  fish.spec.in                share
Doxyfile.help               configure.ac                fish.xcodeproj              src
Doxyfile.user               create_wajig_completions.py install-sh                  tests
Makefile.in                 debian                      lexicon_filter.in
~/fish-shell> autoconf; and ./configure; and make
[...]
fish has now been built.
Use '/Applications/Xcode.app/Contents/Developer/usr/bin/make install' to install fish.
~/fish-shell> ./fish -c 'echo %firefox'
Unable to find a process 'firefox'
fish: echo %firefox
            ^
~/fish-shell> sudo make install
fish has now been built.
Use '/Applications/Xcode.app/Contents/Developer/usr/bin/make install' to install fish.
if test -x install-sh; then true; else chmod 755 install-sh; fi
[...]
[new window]
~> fish --version
fish, version 2.2.0-263-ga02d4dc
~> echo %fish
Unable to find a process 'fish'
fish: echo %fish
            ^
```

**Comment by ridiculousfish** [comment]
OS X is attempting to do Linux-style `/proc` expansion, because the IWYU changes caused us to drop the header that defined KERN_PROCARGS2.

Fixed as fa4ec55c96d3c4233394a90958ea87a0ba90fc91. Thanks for filing this and nice bisecting, it was right on!

**Comment by waterhouse** [comment]
Excellent.  Confirmed working now.
